perfkitbenchmarker/providers/aws/aws_virtual_machine.py

```diff
@@ -465,6 +501,69 @@ def _Create(self):
       self.client_token = str(uuid.uuid4())
       raise errors.Resource.RetryableCreationError()
 
+  def _CreateSpot(self):
+    """Create a Spot VM instance."""
+    placement = OrderedDict()
+    if not util.IsRegion(self.zone):
+      placement['AvailabilityZone'] = self.zone
+    if self.use_dedicated_host:
+      raise errors.Resource.CreationError(
+          'Tenancy=host is not supported for Spot Instances')
+    elif IsPlacementGroupCompatible(self.machine_type):
+      placement['GroupName'] = self.network.placement_group.name
+    block_device_map = GetBlockDeviceMap(self.machine_type,
+                                         self.boot_disk_size,
+                                         self.image,
+                                         self.region)
+    network_interface = [OrderedDict([
+        ('DeviceIndex', 0),
+        ('AssociatePublicIpAddress', True),
+        ('SubnetId', self.network.subnet.id)])]
+    launch_specification = OrderedDict([
+        ('ImageId', self.image),
+        ('InstanceType', self.machine_type),
+        ('KeyName', 'perfkit-key-%s' % FLAGS.run_uri),
+        ('Placement', placement)])
+    if block_device_map:
+      print 'RAW block_device_map'
+      print block_device_map
+      launch_specification['BlockDeviceMappings'] = json.loads(
+          block_device_map, object_pairs_hook=collections.OrderedDict)
+    launch_specification['NetworkInterfaces'] = network_interface
+    create_cmd = util.AWS_PREFIX + [
+        'ec2',
+        'request-spot-instances',
+        '--region=%s' % self.region,
+        '--spot-price=%s' % self.spot_price,
+        '--client-token=%s' % self.client_token,
+        '--launch-specification=%s' % json.dumps(launch_specification,
+                                                 separators=(',', ':'))]
+    print ' '.join(create_cmd)
+    stdout, stderr, _ = vm_util.IssueCommand(create_cmd)
+    create_response = json.loads(stdout)
+    self.spot_instance_request_id =\
+        create_response['SpotInstanceRequests'][0]['SpotInstanceRequestId']
+
+    while True:
+      describe_sir_cmd = util.AWS_PREFIX + [
+          'ec2',
+          'describe-spot-instance-requests',
+          '--spot-instance-request-ids=%s' % self.spot_instance_request_id]
+      stdout, stderr, _ = vm_util.IssueCommand(describe_sir_cmd)
+
+      sir_response = json.loads(stdout)['SpotInstanceRequests']
+      assert len(sir_response) == 1, 'Expected exactly 1 SpotInstanceRequest'
+
+      status_code = sir_response[0]['Status']['Code']
+
+      if status_code in SPOT_INSTANCE_REQUEST_HOLDING_STATUSES or \
+         status_code in SPOT_INSTANCE_REQUEST_TERMINAL_STATUSES:
+        message = sir_response[0]['Status']['Message']
+        raise errors.Resource.CreationError(message)
+      elif status_code == "fulfilled":
+        self.id = sir_response[0]['InstanceId']
```

Comment:
when this succeeds, the next (logged) command is an `aws ec2 describe-instances` so you should know it happens